### ISSUE TITLE
Git issues with `16.0.0-beta.18`: "Host key verification failed. fatal: Could not read from remote repository."

### ISSUE THREAD

**Issue by fernandoespinosa** [semantic-release/semantic-release#1156]
## Current behavior

We're getting some rather frustrating Git connection issues with `16.0.0-beta.18` that do not occur on the stable release `15.13.3`. We need to start using your beta because we need to begin publishing pre-releases from a `beta` branch.

Basically after plugins load, it seems to want to use SSH and ignores the config setting:

```
repositoryUrl: 'https://github.com/PolymathNetwork/polymath-sdk.git'
```

Certainly on the CI server side (TeamCity Build Agent) I am using SSH to fetch and build the sources, but I have the `GITHUB_TOKEN` environment variable properly setup and passed to the build process as well. Indeed, this all works perfectly fined with `15.13.3` but it only breaks once I start using `16.0.0-beta.18`!

## Expected behavior

Everything works perfectly fined with `15.13.3` but it only breaks once I start using `16.0.0-beta.18`!

## Environment

- **semantic-release** version:
```
[18:32:27]	[Step 4/4] yarn run v1.15.2
[18:32:27]	[Step 4/4] $ /opt/buildagent/work/f6a9fa9badf552f8/node_modules/.bin/semantic-release --debug
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.18
```

- CI environment:

```
root@d9ea98f77ce9:/# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
root@d9ea98f77ce9:/# node --version
v11.14.0
root@d9ea98f77ce9:/# yarn --version
1.15.2
```

- Plugins used & **semantic-release** configuration:
```
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.329Z semantic-release:config load config from: /opt/buildagent/work/f6a9fa9badf552f8/package.json
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.338Z semantic-release:config options values: { branches:
[18:32:28]	[Step 4/4]    [ '+([1-9])?(.{+([1-9]),x}).x',
[18:32:28]	[Step 4/4]      'master',
[18:32:28]	[Step 4/4]      'next',
[18:32:28]	[Step 4/4]      'next-major',
[18:32:28]	[Step 4/4]      { name: 'beta', prerelease: true },
[18:32:28]	[Step 4/4]      { name: 'alpha', prerelease: true } ],
[18:32:28]	[Step 4/4]   repositoryUrl: 'https://github.com/PolymathNetwork/polymath-sdk.git',
[18:32:28]	[Step 4/4]   tagFormat: 'v${version}',
[18:32:28]	[Step 4/4]   plugins:
[18:32:28]	[Step 4/4]    [ '@semantic-release/commit-analyzer',
[18:32:28]	[Step 4/4]      '@semantic-release/release-notes-generator',
[18:32:28]	[Step 4/4]      '@semantic-release/npm',
[18:32:28]	[Step 4/4]      '@semantic-release/git',
[18:32:28]	[Step 4/4]      '@semantic-release/github' ],
[18:32:28]	[Step 4/4]   _: [],
[18:32:28]	[Step 4/4]   debug: true,
[18:32:28]	[Step 4/4]   '$0': 'node_modules/.bin/semantic-release' }
```
- CI logs: 

You asked for it:

```
[18:32:27]	[Step 4/4] $ /opt/buildagent/work/f6a9fa9badf552f8/node_modules/.bin/semantic-release --debug
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ℹ  Running semantic-release version 16.0.0-beta.18
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.329Z semantic-release:config load config from: /opt/buildagent/work/f6a9fa9badf552f8/package.json
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.338Z semantic-release:config options values: { branches:
[18:32:28]	[Step 4/4]    [ '+([1-9])?(.{+([1-9]),x}).x',
[18:32:28]	[Step 4/4]      'master',
[18:32:28]	[Step 4/4]      'next',
[18:32:28]	[Step 4/4]      'next-major',
[18:32:28]	[Step 4/4]      { name: 'beta', prerelease: true },
[18:32:28]	[Step 4/4]      { name: 'alpha', prerelease: true } ],
[18:32:28]	[Step 4/4]   repositoryUrl: 'https://github.com/PolymathNetwork/polymath-sdk.git',
[18:32:28]	[Step 4/4]   tagFormat: 'v${version}',
[18:32:28]	[Step 4/4]   plugins:
[18:32:28]	[Step 4/4]    [ '@semantic-release/commit-analyzer',
[18:32:28]	[Step 4/4]      '@semantic-release/release-notes-generator',
[18:32:28]	[Step 4/4]      '@semantic-release/npm',
[18:32:28]	[Step 4/4]      '@semantic-release/git',
[18:32:28]	[Step 4/4]      '@semantic-release/github' ],
[18:32:28]	[Step 4/4]   _: [],
[18:32:28]	[Step 4/4]   debug: true,
[18:32:28]	[Step 4/4]   '$0': 'node_modules/.bin/semantic-release' }
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.603Z semantic-release:plugins options for @semantic-release/npm/verifyConditions: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.605Z semantic-release:plugins options for @semantic-release/git/verifyConditions: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.606Z semantic-release:plugins options for @semantic-release/github/verifyConditions: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/git"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.607Z semantic-release:plugins options for @semantic-release/commit-analyzer/analyzeCommits: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.607Z semantic-release:plugins options for @semantic-release/release-notes-generator/generateNotes: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.608Z semantic-release:plugins options for @semantic-release/npm/prepare: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.608Z semantic-release:plugins options for @semantic-release/git/prepare: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.608Z semantic-release:plugins options for @semantic-release/npm/publish: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/git"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.609Z semantic-release:plugins options for @semantic-release/github/publish: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.610Z semantic-release:plugins options for @semantic-release/npm/addChannel: {}
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.610Z semantic-release:plugins options for @semantic-release/github/success: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[18:32:28]	[Step 4/4] 2019-04-18T18:32:28.615Z semantic-release:plugins options for @semantic-release/github/fail: {}
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[18:32:28]	[Step 4/4] [6:32:28 PM] [semantic-release] › ✖  An error occurred while running semantic-release: { Error: Command failed: git ls-remote --heads origin
[18:32:28]	[Step 4/4] Host key verification failed.
[18:32:28]	[Step 4/4] fatal: Could not read from remote repository.
[18:32:28]	[Step 4/4] 
[18:32:28]	[Step 4/4] Please make sure you have the correct access rights
[18:32:28]	[Step 4/4] and the repository exists.
[18:32:28]	[Step 4/4] 
[18:32:28]	[Step 4/4] 
[18:32:28]	[Step 4/4]     at makeError (/opt/buildagent/work/f6a9fa9badf552f8/node_modules/execa/index.js:174:9)
[18:32:28]	[Step 4/4]     at Promise.all.then.arr (/opt/buildagent/work/f6a9fa9badf552f8/node_modules/execa/index.js:278:16)
[18:32:28]	[Step 4/4]     at processTicksAndRejections (internal/process/task_queues.js:86:5)
[18:32:28]	[Step 4/4]   code: 128,
[18:32:28]	[Step 4/4]   stdout: '',
[18:32:28]	[Step 4/4]   stderr:
[18:32:28]	[Step 4/4]    'Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n',
[18:32:28]	[Step 4/4]   failed: true,
[18:32:28]	[Step 4/4]   signal: null,
[18:32:28]	[Step 4/4]   cmd: 'git ls-remote --heads origin',
[18:32:28]	[Step 4/4]   timedOut: false,
[18:32:28]	[Step 4/4]   killed: false }
```

![image](https://user-images.githubusercontent.com/1607385/56384832-720a5780-61eb-11e9-9a67-39b02d675c10.png)

**Comment by jmodjeski75** [comment]
same issue being discussed in #1170

**Comment by pvdlg** [comment]
Should have been fixed in #1186
Please re-open if not.
